pkg/reconciler/v1alpha1/taskrun/resources/pod.go

```diff
@@ -229,6 +228,14 @@ func TryGetPod(taskRunStatus v1alpha1.TaskRunStatus, gp GetPod) (*corev1.Pod, er
 // MakePod converts TaskRun and TaskSpec objects to a Pod which implements the taskrun specified
 // by the supplied CRD.
 func MakePod(taskRun *v1alpha1.TaskRun, taskSpec v1alpha1.TaskSpec, kubeclient kubernetes.Interface, cache *entrypoint.Cache, logger *zap.SugaredLogger) (*corev1.Pod, error) {
+	// Copy annotations on the build through to the underlying pod to allow users
+	// to specify pod annotations.
+	annotations := map[string]string{}
+	for key, val := range taskRun.Annotations {
+		annotations[key] = val
+	}
+	annotations[ReadyAnnotation] = ""
```

Comment:
can this code be a separate function with unit tests?

Reply:
Good catch.  it turns out there is already a separate function (see `makeAnnotations()` elsewhere in this file) and these few lines are dead code. I've added some unit tests for makeAnnotations.